File: Helpers/TaskSearchMatcher.cs
using Hatch.Models;

namespace Hatch.Helpers;

public static class TaskSearchMatcher
{
    public static bool Matches(TodoItem task, string query) =>
        task.Title.Contains(query, StringComparison.OrdinalIgnoreCase) ||
        (task.Notes != null && task.Notes.Contains(query, StringComparison.OrdinalIgnoreCase)) ||
        task.Tags.Any(t => t.Contains(query, StringComparison.OrdinalIgnoreCase));
}

File: mobile/shared/src/commonMain/kotlin/dev/hatch/sync/TaskSearchMatcher.kt
package dev.hatch.sync

// Transcription of windows/Helpers/TaskSearchMatcher.cs.
object TaskSearchMatcher {

    fun matches(task: TodoItem, query: String): Boolean =
        task.title.contains(query, ignoreCase = true) ||
            (task.notes?.contains(query, ignoreCase = true) == true) ||
            task.tags.any { it.contains(query, ignoreCase = true) }
}
